### PR TITLE
Creating a Jenkins job that will create wrapper jobs 

### PR DESCRIPTION
**Pull request by samuel-rubin**
There are three parts to this:
 - Wrapper_job.groovy
 - Wrapper_variables.yml
 - Wrapper_template

When this is properly configured, Jenkins will have a pipeline with
    the different configurations as Boolean values
The pipeline will then get the configuration values from the yaml file
It will then call the template with those values
The template will then create the jobs to the specification

[skip ci]

Signed-off-by: Samuel Rubin <samuel.rubin@ibm.com>

### CODE REVIEW

**Comment by samuel-rubin** [comment]
@AdamBrousseau

**Comment by samuel-rubin** [comment]
@AdamBrousseau How does this look?

File: buildenv/jenkins/jobs/infrastructure/wrapper_variables.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            Javanext: false
+        string_parameters:
+            PLATFORMS: "aix_ppc-64_cmprssptrs,linux_x86-64,linux_x86-64_cmprssptrs,linux_x86-64_cmprssptrs_cmake,linux_ppc-64_cmprssptrs_le,linux_390-64_cmprssptrs,win_x86-64_cmprssptrs,win_x86,osx_x86-64_cmprssptrs"
+            TEST_TARGETS: "sanity.functional,extended.functional,sanity.system,extended.system"
+            RESTART_TIMEOUT: "10"
+        choice_parameters:
+            BUILD_IDENTIFIER: 
+                - "Nightly"
+                - "Custom"
+            AUTOMATIC_GENERATION:
```
[the remark this answers is not in the document]
Comment (samuel-rubin) [reply buildenv/jenkins/jobs/infrastructure/wrapper_variables.yml]: I'll change the default to true

File: buildenv/jenkins/jobs/infrastructure/wrapper_variables.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+###############################################################################
+
+#================================================================#
+# Specifications for the pipeline job that will be built nightly
+#================================================================#
+Nightly:
+    job_name: "Pipeline-Build-Test-All"
+    job_description: ""
+    build_discarder_logs: 10
```
[the remark this answers is not in the document]
Comment (samuel-rubin) [reply buildenv/jenkins/jobs/infrastructure/wrapper_variables.yml]: Those are the current values for the discarder

File: buildenv/jenkins/jobs/infrastructure/wrapper_variables.yml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            OPENJDK11_REPO:
+                - "https://github.com/ibmruntimes/openj9-openjdk-jdk11.git"
+            OPEMJDK12_REPO:
+                - "https://github.com/ibmruntimes/openj9-openjdk-jdk12.git"
+            OPENJ9_REPO:
+                - "https://github.com/eclipse/openj9.git"
+            OMR_REPO:
+                - "https://github.com/eclipse/openj9-omr.git"
+            LABEL:
+                - "worker"
```
[the remark this answers is not in the document]
Comment (samuel-rubin) [reply buildenv/jenkins/jobs/infrastructure/wrapper_variables.yml]: Since it is the only one that uses that parameter, i'll delete it

**Comment by samuel-rubin** [comment]
I made the changes and here is the my [test job](https://ci.eclipse.org/openj9/view/Sandbox/job/sam-wrapper-test/)

File: buildenv/jenkins/variables/admin_list.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
+###############################################################################
+# Copyright (c) 2018, 2019 IBM Corp. and others
```
[the remark this answers is not in the document]
Comment (AdamBrousseau) [reply buildenv/jenkins/variables/admin_list.yml]: Good catch. How can we proceed now that Sam is finished his term? Is it "ok" if we merge as-is and update it or is there a way to add a commit to this PR?

File: buildenv/jenkins/variables/admin_list.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  - ymanton
+  - zl-wang
+
+infra:
+  - AdamBrousseau
+  - jdekonin
+  - vsebe
+
+extensions:
+  - AdamBrousseau
```
[the remark this answers is not in the document]
Comment (AdamBrousseau) [reply buildenv/jenkins/variables/admin_list.yml]: Not disagreeing, just figured we would keep that list synced with the Github team. I'm ok if we want to remove duplicates.

File: buildenv/jenkins/jobs/infrastructure/wrapper_variables.yml [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+            OMR_BRANCH:
+            PLATFORMS: "ppc64_aix,x86-64_linux,x86-64_linux_xl,ppc64le_linux,s390x_linux,x86-64_windows,x86-32_windows,x86-64_mac"
+            TEST_TARGETS: "sanity.functional,extended.functional,sanity.system,extended.system"
+        choice_parameters:
+            OPENJDK8_REPO:
+                - "https://github.com/ibmruntimes/openj9-openjdk-jdk8.git"
+            OPENJDK11_REPO:
+                - "https://github.com/ibmruntimes/openj9-openjdk-jdk11.git"
+            OPEMJDK12_REPO:
+                - "https://github.com/ibmruntimes/openj9-openjdk-jdk12.git"
```
[the remark this answers is not in the document]
Comment (AdamBrousseau) [reply buildenv/jenkins/jobs/infrastructure/wrapper_variables.yml]: Looks like a typo on `OPEMJDK12` too

**Comment by pshipton** [comment]
@AdamBrousseau you can fetch Sam's commit and then create another commit on Sam's to fix the remaining problems, and push this to your fork. Then create a new PR and we'll close this one.

File: buildenv/jenkins/variables/admin_list.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (pshipton) [reply buildenv/jenkins/variables/admin_list.yml]: If the infra people are committers on extensions, I agree they can stay in the extensions list.

**Comment by pshipton** [comment]
Replaced by  #5638

**Pull request closed without merging**